### PR TITLE
Add context to Ready condition while updating

### PR DESCRIPTION
**Pull request by cmurphy**
When zonal GKE clusters are updating the control plane, the control
plane usually becomes unreachable. This is expected and normal
behavior[1]. However, when the HealthSyncer controller runs, it cannot
reach the API and sets ClusterConditionReady to false with an alarming
error message. This change does not eliminate or ignore the error but
adds more context to the error message to help users understand that
this is normal.

[1] https://cloud.google.com/kubernetes-engine/docs/concepts/cluster-upgrades#upgrading_manually

https://github.com/rancher/rancher/issues/32473

### CODE REVIEW

**Comment by cmurphy** [comment]
Example of the UI change 
image

It's still red and still in "Error" but if you read closely there's some more information

**Review by thedadams — APPROVED** [review]

File: pkg/controllers/management/gke/gke_cluster_handler.go [hunk 1]
```diff
@@ -246,6 +246,18 @@ func (e *gkeOperatorController) onClusterChange(key string, cluster *mgmtv3.Clus
 		e.ClusterEnqueueAfter(cluster.Name, enqueueTime)
 		if failureMessage == "" {
 			logrus.Infof("waiting for cluster GKE [%s] to update", cluster.Name)
+
+			// If the HealthSyncer runs while upgrading a zonal cluster, the control plane may not be reachable.
+			// This adds additional context to the error message to help explain that this is normal.
+			readyMsg := apimgmtv3.ClusterConditionReady.GetMessage(cluster)
```
Comment (rmweir) [inline pkg/controllers/management/gke/gke_cluster_handler.go:252]: Would this message not be found on the updating condition?

**Review by rmweir — CHANGES_REQUESTED** [review]
Are there things we can easily check that would help us determine if it is going through an upgrade?

File: pkg/controllers/management/gke/gke_cluster_handler.go (hunk 1 above)
Comment (cmurphy) [reply pkg/controllers/management/gke/gke_cluster_handler.go:252]: No, it's only the Ready condition that gets the health check message set

https://github.com/rancher/rancher/blob/f5ff8e1fbe76e63d2197185c0d7adda72bb1db70/pkg/controllers/managementuser/healthsyncer/healthsyncer.go#L118
https://github.com/rancher/rancher/blob/f5ff8e1fbe76e63d2197185c0d7adda72bb1db70/pkg/controllers/managementuser/healthsyncer/healthsyncer.go#L80

**Comment by cmurphy** [comment]
> Are there things we can easily check that would help us determine if it is going through an upgrade?

It is checking the gkeclusterconfig phase and only setting this condition during the "updating" phase.

**Review by cbron — APPROVED** [review]
Downside of this is if a cluster really does go down when updating the user may think its ok. But this is a GKE trait, so not much else we can do but tell people that.

**Review by rmweir — APPROVED** [review]
LGTM. My concern was that just checking if it's updating may not be specific enough and appending that message if it isn't upgrading but updating something else may be misleading. This is probably fine as it is assuming the aforementioned doesn't happen often.

**Comment by cmurphy** [comment]
forward port https://github.com/rancher/rancher/pull/32497

**Pull request merged**